src/api/paraswap/index.ts

```diff
@@ -0,0 +1,140 @@
+import { OrderKind } from '@cowprotocol/contracts'
+import { debug, error, warn } from 'loglevel'
+import { ParaSwap, SwapSide } from 'paraswap'
+import { OptimalRate } from 'paraswap-core'
+import { NetworkID, RateOptions } from 'paraswap/build/types'
+import { SupportedChainId } from '../../constants/chains'
+import { CowError } from '../../utils/common'
+import { getTokensFromMarket } from '../../utils/market'
+import {
+  API_NAME,
+  BASE_URL,
+  DEFAULT_RATE_OPTIONS,
+  COMPATIBLE_PARASWAP_CHAINS_WITH_COW,
+  LIB,
+  LOG_PREFIX,
+} from './constants'
+import { ParaswapCowswapNetworkID, ParaswapLibMap, ParaswapPriceQuoteParams, QuoteOptions } from './types'
+import { getParaswapChainId, getValidParams, handleResponse } from './utils'
+
+const ALL_SUPPORTED_CHAIN_IDS = [SupportedChainId.MAINNET]
+
+export default class ParaswapApi {
+  libMap: ParaswapLibMap = LIB
+  name: string
+  apiUrl: string
+  rateOptions: RateOptions
+
+  constructor() {
+    this.name = API_NAME
+    this.apiUrl = BASE_URL
+    this.rateOptions = DEFAULT_RATE_OPTIONS
+  }
+
+  public async getQuote(
+    params: ParaswapPriceQuoteParams & { chainId: NetworkID },
+    options?: Omit<QuoteOptions<true>, 'chainId'>
+  ): Promise<OptimalRate | null>
+  public async getQuote(
+    params: ParaswapPriceQuoteParams & { chainId: ParaswapCowswapNetworkID },
```

Comment:
`ParaswapCowswapNetworkID` overlapping compatible chains, in this case as of rn it's only MAINNET